src/client/remote_generation.py

```diff
@@ -117,19 +151,34 @@ def generate(
                 for constraint in constraints:
                     lm_logits = constraint(last_token_id, lm_logits, hypo_ids)
                 last_token_id, hypo_ids = decoding_algorithm(lm_logits)
-                if seq_idx < inputs.size(1):  # TODO: why is it not a constraint?
+
+                # If samples have padded, so changes only them
```

Comment:
Nit: "samples have padded" sounds a bit confusing (iirc, "to pad" means to apply to something else, not to receive padding)

Suggestion: "If some samples were padded, change only these samples"?